plugins/modules/ovirt_disk.py

```diff
@@ -384,18 +374,35 @@ def transfer(connection, module, direction, transfer_func):
         elif auth.get('ca_file'):
             context.load_verify_locations(cafile=auth.get('ca_file'))
 
-        transfer_connection = HTTPSConnection(
-            destination_url.hostname,
-            destination_url.port,
-            context=context,
-        )
+        try:
+            destination_url = urlparse(transfer.transfer_url)
+            transfer_connection = HTTPSConnection(
+                destination_url.hostname,
+                destination_url.port,
+                context=context,
+            )
+            transfer_func(
+                transfer_service,
+                transfer_connection,
+                destination_url,
+                transfer.signed_ticket
+            )
+        except Exception as e:
+            module.warn("Image transfer fail with message: {}.".format(e))
+            module.warn("Retrying trasfer with proxy.")
+            destination_url = urlparse(transfer.proxy_url)
+            transfer_connection = HTTPSConnection(
+                destination_url.hostname,
+                destination_url.port,
+                context=context,
+            )
```

Comment:
The idea to use the proxy a fallback is good, but the implementation is not.

Lets say you upload 100 GiB image, and the uploads fails after 80 GiB, maybe because storage does not have enough space. This code will fail after sending 80 GiB, and then try to do the same from the proxy, that will fail again.

What we need instead is:

```python
    def create_connection(transfer, context, connect_timeout=10, read_timeout=60):
        url = urlparse(transfer.transfer_url)
        connection = HTTPSConnection(
            url.netloc, context=context, timeout=connect_timeout)
        
        try:
            connection.connect()
        except OSError as e:
            # Typically ConnectionRefusedError or socket.gaierror.
            log.warning(
                "Cannot connect to %s, trying %s: %s",
                transfer.transfer_url, transfer.proxy_url, e)

            url = urlparse(transfer.proxy_url)
            connection = HTTPSConnection(
                url.netloc, context=context, timeout=connect_timeout)
            connection.connect()

        connection.sock.settimeout(read_timeout)
        return connection
```

Once you created a connection, you can use it in the same way with both host and proxy.